libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/service/AuditService.java

```diff
@@ -2,48 +2,76 @@
+import java.util.concurrent.ExecutionException;
 
 import static software.amazon.awssdk.regions.Region.EU_WEST_2;
 import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.SQS_AUDIT_EVENT_QUEUE_URL;
 
 public class AuditService {
-    private final SqsClient sqs;
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final SqsAsyncClient sqs;
     private final String queueUrl;
     private final ObjectMapper objectMapper;
 
-    public AuditService(SqsClient sqs, ConfigService configService) {
+    private List<CompletableFuture<SendMessageResponse>> events = new ArrayList<>();
+
+    public AuditService(SqsAsyncClient sqs, ConfigService configService) {
         this.sqs = sqs;
         this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
         this.objectMapper = new ObjectMapper();
     }
 
-    public AuditService(SqsClient sqs, ConfigService configService, ObjectMapper objectMapper) {
+    public AuditService(
+            SqsAsyncClient sqs, ConfigService configService, ObjectMapper objectMapper) {
         this.sqs = sqs;
         this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
         this.objectMapper = objectMapper;
     }
 
-    public static SqsClient getSqsClient() {
-        return SqsClient.builder()
+    public static SqsAsyncClient getSqsClient() {
+        return SqsAsyncClient.builder()
                 .region(EU_WEST_2)
-                .httpClientBuilder(UrlConnectionHttpClient.builder())
+                .httpClientBuilder(AwsCrtAsyncHttpClient.builder())
                 .build();
     }
 
     public void sendAuditEvent(AuditEvent auditEvent) throws SqsException {
         try {
-            sqs.sendMessage(
-                    SendMessageRequest.builder()
-                            .queueUrl(queueUrl)
-                            .messageBody(objectMapper.writeValueAsString(auditEvent))
-                            .build());
+            var event =
+                    sqs.sendMessage(
+                            SendMessageRequest.builder()
+                                    .queueUrl(queueUrl)
+                                    .messageBody(objectMapper.writeValueAsString(auditEvent))
+                                    .build());
+            events.add(event);
         } catch (JsonProcessingException e) {
             throw new SqsException(e);
         }
     }
+
+    // Await audit events MUST be called before the end of each handler
+    public void awaitAuditEvents() {
+        try {
+            var eventsToAwait = events;
+            events = new ArrayList<>();
+            CompletableFuture.allOf(eventsToAwait.toArray(new CompletableFuture[0])).get();
+        } catch (InterruptedException | ExecutionException e) {
+            LOGGER.error(LogHelper.buildErrorMessage("Failed to send audit event(s)", e));
+            throw new AuditException("Failed to send audit event(s)", e);
```

Comment:
This is a RuntimeException to ease its use - this means an audit failure will fail the lambda, but this is probably a good thing!

I'm in discussion about whether any of the audit events can be marked as non-critical (i.e. we can ignore if they fail), but for now let's assume that they all are, because TICF may need them.